src/operator/numpy/np_dot-inl.h

```diff
@@ -140,17 +141,15 @@ inline void NumpyDotForward(const nnvm::NodeAttrs& attrs,
         Kernel<scalar_mul_kernel<Req>, xpu>::Launch(
           s, out.Size(), out.dptr<DType>(), tensor, scalar);
       });
-    } else if (a_shape.ndim() == 1 || b_shape.ndim() == 1) {
```

Comment:
Why delete this?

Reply:
This was wrong. If a is a vector and b is a tensor, it belongs to case 5 rather than case 4.